go/vt/tabletserver/query_executor.go

```diff
@@ -332,29 +340,27 @@ func (qre *QueryExecutor) execNextval() (*sqltypes.Result, error) {
 			if err != nil {
 				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
 			}
+			// Initialize NextVal if it wasn't already.
+			if t.NextVal == 0 {
+				t.NextVal = nextID
+			}
 			cache, err := qr.Rows[0][1].ParseInt64()
 			if err != nil {
 				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
 			}
 			if cache < 1 {
 				return nil, fmt.Errorf("invalid cache value for sequence %s: %d", qre.plan.TableName, cache)
 			}
-			inc, err := qr.Rows[0][2].ParseInt64()
-			if err != nil {
-				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
-			}
-			if inc < 1 {
-				return nil, fmt.Errorf("invalid increment for sequence %s: %d", qre.plan.TableName, inc)
+			newLast := nextID + cache
+			for newLast <= t.NextVal+inc {
+				newLast += cache
 			}
```

Comment:
Based on offline discussion, there's a risk to using ceil because of possible off-by-epsilon errors with floating point numbers.
It's probably feasible to come up with an expression using integer-only arithmetic. However, the loop is easy to prove correctness, and is something that will rarely get executed because cache will typically be bigger than inc.
TL;DR: We'll leave it as is for now.